src/Nethermind/Nethermind.Trie/ITreeVisitor.cs

```diff
@@ -33,5 +33,9 @@ public interface ITreeVisitor
         void VisitLeaf(TrieNode node, TrieVisitContext trieVisitContext, byte[] value = null);
         
         void VisitCode(Keccak codeHash, TrieVisitContext trieVisitContext);
+
+        bool VisitAccounts => true;
```

Comment:
I am pretty sure it breaks the visitor pattern, visitor should have no knowledge about the structure of what it is visiting or control over visiting mechanism

Reply:
Disagree - visitor knows what it does support, before this change we had a ProofCollector, that would throw on VisitCode and needed to be carefully invoked to avoid that: https://github.com/NethermindEth/nethermind/blob/1d2f21ba04d4fd14cada95479d57b8d368b70af7/src/Nethermind/Nethermind.State/Proofs/ProofCollector.cs#L99

now it declares it doesn't want to visit parts of the tree. This is similar to what ITxTracer does.

If you still think this is bad, can you propose better solution?